File: frontend/src-original-backup/api/recharge.ts
import client from "./client";

export interface RedeemResult {
  success: boolean;
  message: string;
  tickets_added: number;
  balance: number;
}

export interface BalanceResult {
  tickets: number;
  email: string;
}

/** 核销充值卡密 */
export async function redeemCode(code: string): Promise<RedeemResult> {
  const { data } = await client.post("/recharge/redeem", { code });
  return data;
}

/** 查询点券余额 */
export async function getBalance(): Promise<BalanceResult> {
  const { data } = await client.get("/recharge/balance");
  return data;
}

File: backend/app/schemas/recharge.py
"""充值相关 Pydantic 模型"""
from pydantic import BaseModel, Field


class RedeemRequest(BaseModel):
    code: str = Field(..., min_length=1, max_length=64, description="充值卡密")


class RedeemResponse(BaseModel):
    success: bool
    message: str
    tickets_added: int = 0
    balance: int = 0


class BalanceResponse(BaseModel):
    tickets: int
    email: str
